compiler/damlc/tests/src/DamlcTest.hs

```diff
@@ -357,14 +358,15 @@ testsForDamlcTest damlc scriptDar _ = testGroup "damlc test" $
               , "      c <- submit alice $ createCmd T with p = alice"
               , "      submit alice $ exerciseCmd c X"
               ]
-            (exitCode, stdout, _stderr) <-
+            (exitCode, stdout, stderr) <-
```

Comment:
the test that makes this call is flaking on windows, but sadly the test log only says the exit status was wrong (see below). This change will at least show us the contents of stderr upon failure.

```
  damlc test
    Non-existent file:                                         OK (3.11s)
    File with compile error:                                   OK (3.30s)
    Test coverage report and summary:                          OK (9.68s)
    Full test coverage report:                                 FAIL (80.62s)
      compiler\damlc\tests\src\DamlcTest.hs:368:
      expected: ExitSuccess
       but got: ExitFailure 1
      Use -p '$0=="damlc.damlc test.Full test coverage report"' to rerun this test only.
    Full test coverage report with --all set:                  OK (16.70s)
    Filter tests with --test-pattern:                          OK (13.94s)
    Full test coverage report without --all set (but imports): OK (14.09s)
    File with failing script:                                  OK (11.39s)
    damlc test --files outside of project:                     OK (12.86s)
    damlc test --project-root relative:                        OK (15.62s)
    damlc test --project-root proj --junit a.xml:              OK (15.99s)
    grpc-max-message-size applies to service results:          OK (18.08s)
    Rollback archive:                                          OK (13.79s)
    damlc test --files b\B.daml in project:                    OK (15.98s)
    damlc test  in project:                                    OK (10.53s)
```
-- [pipeline](https://dev.azure.com/digitalasset/daml/_build/results?buildId=135905&view=logs&j=800c766b-cf7f-545c-6b0a-ae5b31f2fdcf&t=39c35361-d0fa-5a5b-07df-90fe37b2b8e2) - [test output](https://dev.azure.com/digitalasset/_apis/resources/Containers/11496359/Test%20logs%20Windows?itemPath=Test%20logs%20Windows%2Fcompiler%2Fdamlc%2Ftests%2Fdamlc-test%2Ftest.log)

Reply:
I wonder if there are other places where this might be useful
Perhaps a generic `assertProcessSuccess` could be used to have this behaviour everywhere